src/System.Text.Json/tests/Serialization/DictionaryTests.KeyPolicy.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class DictionaryKeyPolicyTests
```

Comment:
Could you also add a test to verify this works with `JsonExtensionData` dictionary property?

Reply:
And one more to see what the behavior is when you have case insensitive keys + camel case. Just to ensure that the keys are unaffected by the case insentivity

Reply:
> Could you also add a test to verify this works with `JsonExtensionData` dictionary property?

As is, this won't work by default with `JsonExtensionData` dictionary properties. This is consistent with Json.NET. With Json.NET, one needs to specify that naming strategies should apply to extension data: https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_Serialization_NamingStrategy_ProcessDictionaryKeys.htm.

To support this with our options, we might need to add a `ExtensionDataPropertyNamePolicy` or something similar.

Reply:
> And one more to see what the behavior is when you have case insensitive keys + camel case. Just to ensure that the keys are unaffected by the case insentivity

This behavior is covered well in the [PropertyNameTests](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/tests/Serialization/PropertyNameTests.cs).

On a related note, this is the current behavior for dictionary key policies that lead to two or more keys having the same output key (e.g. both myInt and MyInt => myInt):
- On serialization, we write all keys and their values
i.e `new Dictionary<string, int> { { "myInt", 1 }, { "MyInt", 2 } }` => `@"{""myInt"": 1,""myInt"": 2}"`.
This is consistent with Json.NET (at least by default)
- On deserialization, the last value wins
i.e parsing `@"{""myInt"": 1,""MyInt"": 2}"` => `new Dictionary<string, int> { { "myInt", 2 } }`

Reply:
cc @steveharter @ahsonkhan

Reply:
> To support this with our options, we might need to add a ExtensionDataPropertyNamePolicy or something similar.

This was previously discussed - the extension data should keep its original naming. But yes, we could always add another naming policy in the future.

Reply:
@layomia I also have a [PR ](https://github.com/dotnet/corefx/pull/39042) out that deals with extension data. We may have to modify that once your change goes in to ignore the dictionary key policy.